### PR TITLE
vscode-extensions.{matklad.rust-analyzer,vadimcn.vscode-lldb}: fix build error

### PR DESCRIPTION
**Pull request by oxalica**
###### Motivation for this change
Workaround for #100077 to fix build errors on current master.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/rust/rust-analyzer/update.sh [hunk 1]
```diff
@@ -51,6 +51,10 @@ build_deps="../../../../misc/vscode-extensions/rust-analyzer/build-deps"
 jq '{ name, version, dependencies: (.dependencies + .devDependencies) }' "$node_src/package.json" \
     >"$build_deps/package.json.new"
 
+# FIXME: Workaround for https://github.com/NixOS/nixpkgs/issues/100077
```
Comment (midchildan) [inline pkgs/development/tools/rust/rust-analyzer/update.sh:54]: Could you also add a comment explaining how this workaround solves the problem?

**Review by midchildan — CHANGES_REQUESTED** [review]

File: pkgs/development/tools/rust/rust-analyzer/update.sh (hunk 1 above)
Comment (oxalica) [reply pkgs/development/tools/rust/rust-analyzer/update.sh:54]: Sorry but I don't know why it works either. :disappointed:

`node2nix` generates different overrides on this package when explicitly specified, and the default one makes npm unhappy.
See also https://github.com/svanderburg/node2nix/issues/171#issuecomment-707289163

Comment (midchildan) [reply pkgs/development/tools/rust/rust-analyzer/update.sh:54]: I see. In that case, could you also add a link to https://github.com/svanderburg/node2nix/issues/171#issuecomment-707289163 in the comment?

**Review by midchildan — APPROVED** [review]

**Review by midchildan — APPROVED** [review]

**Comment by oxalica** [comment]
ping @jonringer

**Comment by oxalica** [comment]
Close for:
- `vscode-extensions.matklad.rust-analyzer` is successfully built on master for no reason. :confused: 
- `vscode-lldb` fix is included in #101179.

**Pull request closed without merging**